ast/functions.go

```diff
@@ -531,6 +537,39 @@ func (n *AggregateFuncExpr) updateSum() error {
 	return nil
 }
 
+func (n *AggregateFuncExpr) updateGroupConcat() error {
+	ctx := n.GetContext()
+	vals := make([]interface{}, 0, len(n.Args))
+	for _, a := range n.Args {
+		value := a.GetValue()
+		if value == nil {
+			return nil
+		}
+		vals = append(vals, value)
+	}
+	if n.Distinct {
+		d, err := ctx.distinctChecker.Check(vals)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !d {
+			return nil
+		}
+	}
+	var buf bytes.Buffer
+	if ctx.Value != nil {
+		// now use comma separator
+		buf.WriteString(ctx.Value.(string))
+		buf.WriteString(",")
+	}
+	for _, val := range vals {
+		buf.WriteString(fmt.Sprintf("%v", val))
```

Comment:
Yes, it is the same way with MySQL.
mysql> select * from tab0;
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    59
Current database: test

+------+------+------+
| col0 | col1 | col2 |
+------+------+------+
|    1 |    2 |    3 |
|    2 |    1 |    3 |
|    3 |    1 |    2 |
|    4 |    5 |    1 |
|    6 |    3 |    1 |
+------+------+------+
5 rows in set (0.02 sec)

mysql> select group_concat(col0, col1) from tab0 group by col2;
+--------------------------+
| group_concat(col0, col1) |
+--------------------------+
| 45,63                    |
| 31                       |
| 12,21                    |
+--------------------------+
3 rows in set (0.00 sec)

mysql>